pkg/wait/conditions.go

```diff
@@ -227,80 +228,89 @@ func IsAppVersion(ctx context.Context, kubeClient *client.Client, appName string
 
 // IsClusterConditionSet returns a WaitCondition that checks if a Cluster resource has the specified condition with the expected status.
 func IsClusterConditionSet(ctx context.Context, kubeClient *client.Client, clusterName string, clusterNamespace string, conditionType capi.ConditionType, expectedStatus corev1.ConditionStatus, expectedReason string) WaitCondition {
-	cluster := &capi.Cluster{
-		ObjectMeta: v1.ObjectMeta{
-			Name:      clusterName,
-			Namespace: clusterNamespace,
-		},
-	}
+	return func() (bool, error) {
+		cluster := &capi.Cluster{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      clusterName,
+				Namespace: clusterNamespace,
+			},
+		}
+		if err := kubeClient.Client.Get(ctx, cr.ObjectKeyFromObject(cluster), cluster); err != nil {
+			return false, err
+		}
```

Comment:
What was the reason for pulling this out and duplicating the code instead of keeping it in the shared function?